### PR TITLE
utils: integrate support for caching when building

### PR DESCRIPTION
**Pull request by compnerd**
Integrate support for sccache into the build system.  This allows us to get a reasonable performance increase for clean builds for the C/C++ side of the build.  This is currently limited to the MSVC compiler as the extended compiler flags cause issues for sccache.

### CODE REVIEW

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by compnerd** [comment]
@swift-ci please test

File: utils/build.ps1 [hunk 1]
```diff
@@ -26,9 +26,6 @@ The path to a directory that mimics a file system image root,
 under which "Library" and "Program Files" subdirectories will be created
 with the files installed by CMake.
 
-.PARAMETER BuildType
```
Comment (etcwilde) [inline utils/build.ps1:29]: Is the removal of debug builds related here? sccache should see the flag in the compiler invocation and realize that it's a different object being emitted.

Comment (compnerd) [reply utils/build.ps1:29]: Yes, it is related.  The Swift build system is just not able to support this (sccache + PDBs don't work very well together, the other changes to use the embedded debug is related).  Once we upgrade to a reasonable version of CMake, we should be able to use the `CMAKE_MSVC_DEBUG_INFORMATION_FORMAT=Embedded` and get the build types to work again.

Comment (etcwilde) [reply utils/build.ps1:29]: hmm, okay :/

**Review by etcwilde — APPROVED** [review]

**Comment by compnerd** [comment]
@swift-ci please build toolchain Windows platform

**Pull request merged**